projects/ks89/angular-modal-gallery/src/lib/components/previews/previews.component.ts

```diff
@@ -251,80 +235,51 @@ export class PreviewsComponent extends AccessibleComponent implements OnInit, On
    * @param InternalLibImage[] images is the array of all images.
    */
   private initPreviews(currentImage: InternalLibImage, images: InternalLibImage[]): void {
-    let index: number;
-    try {
-      index = getIndex(currentImage, images);
-    } catch (err) {
-      throw err;
-    }
-    switch (index) {
-      case 0:
-        // first image
-        this.setBeginningIndexesPreviews();
-        break;
-      case images.length - 1:
-        // last image
-        this.setEndIndexesPreviews();
-        break;
-      default:
-        // other images
-        this.setIndexesPreviews();
-        break;
-    }
+    this.setIndexesPreviews(currentImage, images);
     this.previews = images.filter((img: InternalLibImage, i: number) => i >= this.start && i < this.end);
   }
 
-  /**
-   * Private method to init both `start` and `end` to the beginning.
-   */
-  private setBeginningIndexesPreviews(): void {
-    if (!this.previewConfig || !this.images) {
-      throw new Error('Internal library error - previewConfig and images must be defined');
-    }
-    this.start = 0;
-    this.end = Math.min(this.previewConfig.number as number, this.images.length);
-  }
-
-  /**
-   * Private method to init both `start` and `end` to the end.
-   */
-  private setEndIndexesPreviews(): void {
-    if (!this.previewConfig || !this.images) {
-      throw new Error('Internal library error - previewConfig and images must be defined');
-    }
-    this.start = this.images.length - 1 - ((this.previewConfig.number as number) - 1);
-    this.end = this.images.length;
-  }
-
   /**
    * Private method to update both `start` and `end` based on the currentImage.
    */
-  private setIndexesPreviews(): void {
-    if (!this.previewConfig || !this.images || !this.currentImage) {
+  private setIndexesPreviews(currentImage: InternalLibImage, images: InternalLibImage[]): void {
+    if (!this.previewConfig || !images || !currentImage) {
```

Comment:
All the indexes calculation logic happens here, for a given currentImage.
The oscillation of previews number was happening mostly because start and end were calculted separately. In the new logic, there are co-dependent. Also, an effort was added to make sure start and end also stay in the boundaries [0, nbImages]